pkg/kubectl/scale.go

```diff
@@ -53,7 +53,10 @@ type Scaler interface {
 	ScaleSimple(namespace, name string, preconditions *ScalePrecondition, newSize uint) (updatedResourceVersion string, err error)
 }
 
-func ScalerFor(kind schema.GroupKind, c internalclientset.Interface) (Scaler, error) {
+// ScalerFor gets a scaler for a given resource
+// TODO(p0lyn0mial): remove one of the case and let it fall through to GenericScaler
```

Comment:
before I remove one of the case I want to see the results of e2e tests.

Reply:
> before I remove one of the case I want to see the results of e2e tests.

Ok.  I think I'd like to see one case removed before we go through the violence of this refactor.  I understand the intermediate state, but I'd like to show progress before we push it all out.